packages/fx-core/src/common/m365/launchHelper.ts

```diff
@@ -26,15 +26,27 @@ export class LaunchHelper {
     hub: HubTypes,
     teamsAppId: string,
     capabilities: string[],
-    withLoginHint = true
+    withLoginHint = true,
+    isCopilotPlugin = false
   ): Promise<Result<string, FxError>> {
     const loginHint = withLoginHint
       ? (await this.getUpnFromToken()) ?? "login_your_m365_account" // a workaround that user has the chance to login
       : undefined;
     let url: URL;
     switch (hub) {
       case HubTypes.teams: {
-        const baseUrl = `https://teams.microsoft.com/l/app/${teamsAppId}?installAppPackage=true&webjoin=true`;
+        let installAppPackage = true;
+        if (
+          isCopilotPlugin &&
+          !capabilities.includes("staticTab") &&
+          !capabilities.includes("configurableTab") &&
+          !capabilities.includes("Bot")
+        ) {
+          installAppPackage = false;
+        }
+        const baseUrl = installAppPackage
+          ? `https://teams.microsoft.com/l/app/${teamsAppId}?installAppPackage=true&webjoin=true`
+          : "https://teams.microsoft.com";
```

Comment:
webjoin?

Reply:
I tried `webjoin` but found that it worked for `l/app/${teamsAppId}?installAppPackage=true` but took no effect for `https://teams.microsoft.com`, i.e. the below window will not be poped out without `webjoin` or even with `webjoin=false`.
![image](https://github.com/OfficeDev/TeamsFx/assets/37978464/4615aef3-ee9b-4332-b558-2515eec4f63e)